src/coreclr/src/jit/morph.cpp

```diff
@@ -17401,6 +17405,9 @@ void Compiler::fgPromoteStructs()
     lvaStructPromotionInfo structPromotionInfo;
     bool                   tooManyLocalsReported = false;
 
+    // Clear the structPromotionHelper, since it is conservative about looking up SIMD info.
```

Comment:
I think I am not following it. Could you please explain?  `structPromotionHelper` has `structPromotionInfo` that is a private temporary buffer that we use to keep information about the last analyzed type. We access it only in `TryPromoteStructVar` and nowhere else. 
Where could we access  `structPromotionHelper->structPromotionInfo.typeHnd` that is cleared here?

Reply:
The cache is used by the inliner, during which it is conservative about looking up SIMD info. I clarified the comment a bit.